### PR TITLE
Add parser ansible_tower_custom

### PR DESCRIPTION
**Pull request by wushiqinlou**
Signed-off-by: jiazhang <jiazhang@redhat.com>

**Please do not include links to any websites that are not publicly accessible. You may include non-link reference numbers to help you and your team identify non-public references.**

### All Pull Requests:

* [x] Have you followed the guidelines in our Contributing document?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Issue/Bug Fix:
Provide complete details of the issue and how this PR fixes the issue. You can link to one or more existing open, publicly-accessible issue(s) that provide details.

### Enhancement:
Provide complete details of the enhancement.

### CODE REVIEW

File: insights/parsers/ansible_tower_config_custom.py
Comment (bfahr) [inline insights/parsers/ansible_tower_config_custom.py:11]: @wushiqinlou filters should only be added to a parser if they are necessary for the parser to work correctly.  For instance if you are parsing a INI-type config file with headings like `[section_name]` then you would need to add a filter to the parser `[` to ensure that the config file parser won't raise an exception.  In this case rules using this parser should be adding any necessary filters.  

Also, `=` is not a good filter since it will collect too many lines from the file.

File: insights/parsers/ansible_tower_config_custom.py
Comment (bfahr) [inline insights/parsers/ansible_tower_config_custom.py:15]: Since this parser will be storing key/value pairs it really should be derived from `dict` instead of `LegacyItemAccess`.  Here is [an example](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/partitions.py) a parser implementing `dict`.

**Review by bfahr — CHANGES_REQUESTED** [review]
@wushiqinlou also please complete the description for your enhancement to insights core.  The description is necessary for us to fully evaluate your PR.

File: insights/parsers/ansible_tower_config_custom.py
Comment (wushiqinlou) [reply insights/parsers/ansible_tower_config_custom.py:11]: @bfahr - In the former code, I checked all of the content without filter. However, per discussion with @xiangce , the file "/etc/tower/conf.d/custom.py" is python file, it is possible that the content format will change, then we will get unexpected result. As the new rule only needs to check option "AWX_CLEANUP_PATHS" now and sensitive information may be configured in such file, we decide to add filter into this parser.

File: insights/parsers/ansible_tower_config_custom.py
Comment (wushiqinlou) [reply insights/parsers/ansible_tower_config_custom.py]: @bfahr - Good to me.

**Review by xiangce — APPROVED** [review]
Looks good to me now. Approve pending approval of spec.

File: insights/parsers/ansible_tower_config_custom.py
Comment (bfahr) [reply insights/parsers/ansible_tower_config_custom.py]: I agree that the spec needs to be filtered, but my recommendation is that the filter should only be added by rules using the parser and not directly by the parser.  Parsers should be general purpose and by adding that filter you are making it specific to a particular rule.  The only parsers that should be adding filters for a filtered spec are those that require certain information to function properly as I described above in the INI file parser example.

Comment (wushiqinlou) [reply insights/parsers/ansible_tower_config_custom.py]: @bfahr - ACK, removed the filter from the parser.

File: insights/specs/__init__.py
Comment (bfahr) [reply insights/specs/__init__.py:22]: I'll fix this since it was already merged.

Comment (wushiqinlou) [reply insights/specs/__init__.py:22]: @bfahr - yes, we need to add `multi_output=True`. Thanks for your update!

Comment (xiangce) [reply insights/specs/__init__.py:22]: @bfahr - Sorry for bringing this trouble.

**Pull request merged**
